### PR TITLE
Improve statistics collection

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🎯 Goal

Improve the statistics collection WebRTC layer and align implementation with other clients.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamVideo/Models/CallStatsReport.swift [hunk 1]
```diff
@@ -24,13 +24,20 @@ public struct CallStatsReport: Sendable {
 }
 
 /// A struct representing statistics for participants in the call.
-public struct ParticipantsStats: Sendable {
+public struct ParticipantsStats: Sendable, Equatable {
     /// The report containing statistics for individual participants.
-    public let report: [String: BaseStats]
+    public let report: [String: [BaseStats]]
```
[the remark this answers is not in the document]
Comment (ipavlidakis) [reply Sources/StreamVideo/Models/CallStatsReport.swift:29]: I don't foresee anything major. Those are very small objects anyway. We can keep it in our minds for memory spikes

File: StreamVideoTests/WebRTC/Statistics/StreamCallStatisticsReporter_Tests.swift [hunk 2]
```diff
@@ -0,0 +1,5 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
```
[the remark this answers is not in the document]
Comment (ipavlidakis) [reply StreamVideoTests/WebRTC/Statistics/StreamCallStatisticsReporter_Tests.swift:5]: My initial plan was to but as it acts as a coordinator and its internals are not exposed i end up forfeiting this idea. Let me remove the file

File: Sources/StreamVideo/Models/CallStatsReport.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (ipavlidakis) [reply Sources/StreamVideo/Models/CallStatsReport.swift:29]: Actually the count of those can be estimated. We get 3 baseStats for the publisher (1 for each stream) and 1 for every other participant. So even in a call with 1000 users we will 1002 items. Each of those items has 11 numeric values (8 bytes) and 3 strings (which don't tend to be very long). A rough estimate is 150 bytes per item which goes around 146Kb

[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamVideo/Models/CallStatsReport.swift:29]: nevermind, mixed it up with the whole stats report - the raw stats are not part of this.

**Pull request merged**